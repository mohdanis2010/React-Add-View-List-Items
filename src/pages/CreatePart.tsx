import React, { useState } from "react";
import Button from "@material-ui/core/Button";
import TextField from "@material-ui/core/TextField";
import "../css/list-style.css";

const CreatePart = ({ addTodo }: { addTodo: any }) => {
  const [name, setName] = useState("");
  const [status, setStatus] = useState("");

  const handleSubmit = (e: any) => {
    e.preventDefault();
    if (!name) return;
    addTodo(name, status);
    setName("");
    setStatus("");
  };

  return (
    <form>
      <div className="input-box">
        <TextField
          id="nametatusText"
          label="Enter Name"
          defaultValue="Name Value"
          value={name}
          onChange={(e) => setName(e.target.value)}
          data-testid="name"
        />
      </div>
      <div className="input-box">
        <TextField
          id="statusText"
          label="Enter Status"
          defaultValue="Status Value"
          value={status}
          onChange={(e) => setStatus(e.target.value)}
          className="input-box"
        />
      </div>
      <div className="btn-wrapper">
      <Button variant="contained" color="primary" onClick={handleSubmit}>
        Add Item
      </Button>
      </div>
    </form>
  );
};

export default CreatePart;
