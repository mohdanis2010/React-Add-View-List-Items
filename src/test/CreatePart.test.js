import React from "react";
import {
  render,
  cleanup,
  fireEvent,
  waitForElement,
} from "@testing-library/react";
import CreatePart from "../pages/CreatePart";

afterEach(cleanup);

it("should take a snapshot", () => {
  const { asFragment } = render(<CreatePart />);
  expect(asFragment(<CreatePart />)).toMatchSnapshot();
});

it("input on change value", () => {
    const { getByTestId } = render(<CreatePart />);
    fireEvent.input(getByTestId("name"), { target: { value: "12" } });
  });