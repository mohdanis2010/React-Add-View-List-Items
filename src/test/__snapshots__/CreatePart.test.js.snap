// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      class=""
      data-testid="name"
      id="name"
      type="text"
      value=""
    />
    <input
      class=""
      type="text"
      value=""
    />
    <button>
      Add Item
    </button>
  </form>
</DocumentFragment>
`;
